src/sentry_plugins/github/endpoints/webhook.py

```diff
@@ -295,20 +295,17 @@ def __call__(self, event, organization):
             repo.config['name'] = event['repository']['full_name']
             repo.save()
 
-        if action == 'opened':
-            self._handle_created(event, organization, repo, is_apps)
-        if action == 'edited' or action == 'closed':
-            self._handle_updated(event, organization, repo)
-
-    def _handle_created(self, event, organization, repo, is_apps):
-        """PR was created"""
-
         pull_request = event['pull_request']
         number = pull_request['number']
         title = pull_request['title']
         body = pull_request['body']
         user = pull_request['user']
-        merge_commit_sha = pull_request['merge_commit_sha']
+        # The value of the merge_commit_sha attribute changes depending on the state of the pull request. Before a pull request is merged, the merge_commit_sha attribute holds the SHA of the test merge commit. After a pull request is merged, the attribute changes depending on how the pull request was merged:
+        # - If the pull request was merged as a merge commit, the attribute represents the SHA of the merge commit.
+        # - If the pull request was merged via a squash, the attribute represents the SHA of the squashed commit on the base branch.
+        # - If the pull request was rebased, the attribute represents the commit that the base branch was updated to.
+        # https://developer.github.com/v3/pulls/#get-a-single-pull-request
+        merge_commit_sha = pull_request['merge_commit_sha'] if pull_request['merged'] else None
```

Comment:
could you help me understand why this improves the behavior?

Reply:
I think we want to create a resolved_in link when the PR is opened, not when the pr is closed, 
(and update that link if the merged_sha changes)

is my understanding correct that this would be None until the merge happens? Does that stil mark the issue as resolved or create an activity item?

Reply:
@MaxBittker this field i think 1) isnt filled except on close, 2) wont be correct until close